armi/reactor/blueprints/reactorBlueprint.py

```diff
@@ -118,89 +116,107 @@ def _resolveSystemType(typ: str):
-            # because they will be populated with DistributeState.
+            # Non-primary nodes get the reactor via DistributeState.
             return None
 
-        # TODO: This is also pretty specific to Core-like things. We envision systems
-        # with non-Core-like structure. Again, probably only doable with subclassing of
-        # Blueprints
-        if loadAssems and gridDesign is not None:
-            self._loadAssemblies(cs, system, gridDesign.gridContents, bp)
-
-            # TODO: This post-construction work is specific to Cores for now. We need to
-            # generalize this. Things to consider:
-            # - Should the Core be able to do geom modifications itself, since it already
-            # has the grid constructed from the grid design?
-            # - Should the summary be so specifically Material data? Should this be done for
-            # non-Cores? Like geom modifications, could this just be done in processLoading?
-            # Should it be invoked higher up, by whatever code is requesting the Reactor be
-            # built from Blueprints?
-            if isinstance(system, reactors.Core):
+        system = self._constructComposites(cs, bp, loadComps, system, gridDesign)
+
+        return system
+
+    def _constructComposites(self, cs, bp, loadComps, system, gridDesign):
+        """Fill a grid with composities, if there are any to fill.
+
+        Parameters
+        ----------
+        cs : Settings object.
+            armi settings to apply
+        bp : Blueprints object.
+            armi blueprints to apply
+        loadComps : bool
+            whether to fill reactor with composities, as defined in blueprints, or not
+        system : Composite
+            The composite we are building.
+        gridDesign : GridBlueprint
+            The defintion of the grid on the object.
+
+        Returns
+        -------
+        Composite
+            A Composite object with a grid, like a Spent Fuel Pool or other ex-core structure.
+        """
+        from armi.reactor.reactors import Core  # avoid circular import
+
+        if loadComps and gridDesign is not None:
+            self._loadComposites(cs, system, gridDesign.gridContents, bp)
+
+            if isinstance(system, Core):
                 summarizeMaterialData(system)
                 self._modifyGeometry(system, gridDesign)
                 system.processLoading(cs)
+
         return system
 
-    def _loadAssemblies(self, cs, container, gridContents, bp):
-        runLog.header(
-            "=========== Adding Assemblies to {} ===========".format(container)
-        )
+    def _loadComposites(self, cs, container, gridContents, bp):
+        runLog.header(f"=========== Adding Composites to {container} ===========")
         badLocations = set()
         for locationInfo, aTypeID in gridContents.items():
+
+            # TODO: We should allow for non-Assembly objects/geometries to be loaded into the grid.
```

Comment:
NOTE: I am not adding a new today. This section of this file had 2 TODOs, I rephrased and reduced it to 1.